src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,222 @@
+<script context="module" lang="ts">
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-segmented-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
+
+          value?: string
+          children?: any
+        }
+      }
+    }
+  }
+</script>
+
+<script lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements'
+  import { createEventDispatcher, onMount } from 'svelte'
+
+  type Size = 'default' | 'small' | 'tiny'
+
+  export let selected: string | undefined = undefined
+  export let size: Size = 'default'
+
+  let segmentedControl: HTMLDivElement;
+
+  let pillWidth: number;
+  let pillPosition: number;
+
+  $: console.log(selected)
+
+  const dispatch = createEventDispatcher<{
+    'select-item': any
+  }>()
+
+  function getValue(e: Element) {
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    return e.getAttribute('value') ?? e['value'] ?? e.textContent
+  }
+
+  $: controlItems = Array.from(
+    // TODO(petemil): support slot *descendents* that are leo-menu-item or leo-option (or something else unstyled like leo-menu-action?) so that we can
+    // select items that are in complex positions (see an example in the browser app menu zoom controls).
+    (segmentedControl?.querySelector('slot') as HTMLSlotElement)
+      ?.assignedElements()
+      ?.filter((element) =>
+        ['LEO-SEGMENTED-ITEM'].includes(element.tagName)
+      ) ??
+      segmentedControl?.querySelectorAll(
+        ':is(leo-segmented-item)'
+      ) ??
+      []
+  ) as HTMLElement[]
+
+  $: {
+    for (const controlItem of controlItems) {
+      controlItem.setAttribute('tabindex', '0')
+
+      if (controlItem.tagName === 'LEO-SEGMENTED-ITEM') {
+        controlItem.setAttribute('role', 'option')
+
+        if (selected === getValue(controlItem)) {
+          controlItem.setAttribute('aria-selected', '')
+          pillWidth = controlItem.clientWidth;
+          pillPosition = controlItem.offsetLeft;
+        } else controlItem.removeAttribute('aria-selected')
+      } else {
+        controlItem.setAttribute('role', 'menuitem')
+      }
+    }
+  }
+
+  function selectItem(e: Event) {
+    // Find the option which was clicked on, if any.
+    const item = controlItems.find((item) => e.composedPath().includes(item))
+
+    // If the event was triggered for something which isn't an option don't fire
+    // a change event.
+    if (!item) return
+
+    // Use data-is-interactive=true to prevent the menu from closing when selected. This implies
```

Comment:
I don't think this comment applies, does it? We don't support interactivity inside our segmented control.